cmake/IgnSetCompilerFlags.cmake

```diff
@@ -115,11 +115,18 @@ endmacro()
 # Set up compilation flags for GCC or Clang
 macro(ign_setup_gcc_or_clang)
 
+  option(USE_DEFAULT_VISIBILITY_HIDDEN "Hide symbols by default if they are not explicitly specified as visible" FALSE)
```

Comment:
I think we should make this on by default in the next major version. I think many of other ignition packages also assume it's hidden by default, since they are using `_VISIBLE` macros instead of `_HIDDEN` macros.

Reply:
Agreed. I don't think we should close #166 until it becomes the default.